Storage/ThirdParty/SwiftData.swift

```diff
@@ -92,10 +100,30 @@ open class SwiftData {
             if self.sharedConnection == nil {
                 log.debug(">>> Creating shared SQLiteDBConnection for \(self.filename) on thread \(Thread.current).")
                 self.sharedConnection = ConcreteSQLiteDBConnection(filename: self.filename, flags: SwiftData.Flags.readWriteCreate.toSQL(), key: self.key, prevKey: self.prevKey)
+                didCreateConnection = true
             }
             connection = self.sharedConnection
         }
 
+        // Should only be `true` when the `ConcreteSQLiteDBConnection` is first created. This
+        // happens when the database is accessed for the very first time. It should also happen
+        // again when the database is first accessed after being re-opened. Either way, this is
+        // the moment when we want to attempt to setup our connection (i.e. create the schema).
+        if didCreateConnection {
+            // If there is no callback for setting up the connection, it means we have already
+            // succeeded in setting it up and there is no need to re-run it.
+            guard let connectionCreatedCallback = self.connectionCreatedCallback else {
+                return connection
+            }
+
+            // Run our callback to set up the the connection (i.e. create the schema). If it
+            // succeeds (returns `true`), we can forget it so we don't attempt to re-run it
+            // later if the connection gets re-opened.
+            if connectionCreatedCallback() {
+                self.connectionCreatedCallback = nil
```

Comment:
Doing this is an attempt to cut the reference cycle.

However, we _should_ re-run the callback when reopening the connection! What if we enter the background and an extension wipes the DB? A `user_version` check is very, very cheap; we should feel free to do it each time we open a connection.

The problem then becomes one of avoiding potential memory leaks from a cycle; see notes above.